src/components/IncompleteSubReqCourse.vue

```diff
@@ -1 +1,186 @@
+  scrollable: boolean;
+  displayCourses: boolean;
+}
+
+export default Vue.extend({
+  mounted() {
+    this.$el.addEventListener('touchmove', this.dragListener, { passive: false });
+    // @ts-ignore
+    const service = Vue.$dragula.$service;
+    service.eventBus.$on('drag', () => {
+      this.scrollable = true;
+    });
+    service.eventBus.$on('drop', () => {
+      this.scrollable = true;
+    });
+
+  },
+  data() : Data {
+    return {
+      courseObjects: [],
+      scrollable: false,
+      displayCourses: false
+    };
+  },
+  beforeDestroy() {
+    this.$el.removeEventListener('touchmove', this.dragListener);
+  },
+  props: {
+    subReq: Object as PropType<DisplayableRequirementFulfillment>,
+    subReqCourseId: Number,
+    crseInfoObjects: Array as PropType<CrseInfo[]>,
+    subReqCourseObjectsNotTakenArray: Array as PropType<AppCourse[]>,
+    dataReady: Boolean,
+    displayDescription: Boolean
+  },
+  watch: {
+    dataReady: {
+      immediate: true,
+      handler(dataReady) {
+        if (dataReady && this.subReqCourseObjectsNotTakenArray.length > 0) {
+          this.getFirstFourCourseObjects();
+          this.displayCourses = true;
+        }
+      }
+    }
+  },
+  computed: {
+    addCourseLabel() {
+      let label = 'Add Course';
+      if (this.subReq.fulfilledBy === 'courses') {
+        label = `Add Course ${this.subReqCourseId + 1}`;
+      }
+      return label;
+    },
+    seeAll() {
+      return 'See all >';
+    },
+    showSeeAllLabel() {
+      // Only show See all label when there are more than 4 courses
+      const allCrseIds = this.crseInfoObjects.map(crseInfoObject => crseInfoObject.crseIds).flat();
+      return allCrseIds.length > 4;
+    }
+  },
+  methods: {
+    dragListener(event: { preventDefault: () => void; }) {
+      if (!this.$data.scrollable) event.preventDefault();
+    },
+    getFirstFourCourseObjects() {
+      let firstFourCourseObjects: AppCourse[] = [];
+      for (let i = 0; firstFourCourseObjects.length < 4 && i < this.crseInfoObjects.length; i += 1){
+        const crseInfoObject = this.crseInfoObjects[i];
+        const filteredCourses: AppCourse[] = this.subReqCourseObjectsNotTakenArray.filter(course => crseInfoObject.crseIds.includes(course.crseId));
+        const numRemainingCourses = Math.min(4 - firstFourCourseObjects.length, filteredCourses.length);
+        firstFourCourseObjects = firstFourCourseObjects.concat(filteredCourses.slice(0, numRemainingCourses));
+      }
+      this.courseObjects = firstFourCourseObjects;
+    }
+  }
+});
+</script>
+
+<style scoped lang="scss">
+@import "@/assets/scss/_variables.scss";
+
+.separator {
+  height: 1px;
+  width: 100%;
+  background-color: #d7d7d7;
+}
+
+.draggable-requirements {
+  &-wrapper {
+    margin-top: 2%;
+    margin-bottom: 2%;
+  }
+  &-heading {
+    display: flex;
+    justify-content: space-between;
+    &-label{
+      font-size: 14px;
+      line-height: 17px;
+      color: #9E9E9E;
```

Comment:
Haven't checked but, is this color or should this color be in our _variables.scss?

Reply:
At the moment, the color doesn't exist. I think that's a good idea since this color will be used for labels on complete sub req courses. I'll make a variable for this color and the color of the separator